overlord/devicestate/handlers_install.go

```diff
@@ -313,25 +315,43 @@ func checkEncryption(model *asserts.Model) (res bool, err error) {
 		return false, nil
 	}
 
-	// encryption is required in secured devices and optional in other grades
-	if err := secbootCheckKeySealingSupported(); err != nil {
+	// check if the model prefers to be unencrypted
+	// TODO: provide way to select via install chooser menu
+	//       if the install is unencrypted or encrypted
+	if model.StorageSafety() == asserts.StorageSafetyPreferUnencrypted {
+		logger.Noticef(`installing system unencrypted to comply with prefer-unencrypted storage-safety model option`)
+		return false, nil
+	}
+
+	// check if encryption is available
+	var (
+		hasFDESetupHook    bool
+		checkEncryptionErr error
+	)
+	if kernelInfo, err := snapstate.KernelInfo(st, deviceCtx); err == nil {
```

Comment:
is there a scenario in which snapstate.KernelInfo(st, deviceCtx) can fail and the install still proceeds?

Reply:
Not in real code AFAICT, in the tests there is.